ICSharpCode.Decompiler/IL/ILReader.cs

```diff
@@ -400,6 +405,10 @@ void ReadInstructions(CancellationToken cancellationToken)
 						currentStack = ImmutableStack<ILVariable>.Empty;
 					}
 				}
+
+				if (IsSequencePointInstruction(decodedInstruction)) {
```

Comment:
Could use `IsSequencePointInstruction(decodedInstruction) || startedWithEmptyStack` here.
Where you could initialize `bool startedWithEmptyStack = currentStack.IsEmpty;` before the `DecodeInstruction()` call.